src/pyk/kore/rpc.py

```diff
@@ -368,13 +368,35 @@ def __init__(self, error: str, context: Iterable[str]):
 
 @final
 @dataclass
-class ModuleError(KoreClientError):
+class UnknownModuleError(KoreClientError):
     module_name: str
 
     def __init__(self, module_name: str):
         self.module_name = module_name
         super().__init__(f'Could not find module: {self.module_name}')
 
+@final
+@dataclass
+class InvalidModuleError(KoreClientError):
+    error: str
+    context: tuple[str, ...] | None
+
+    def __init__(self, error: str, context: Iterable[str] | None):
+        self.error = error
+        self.context = tuple(context) if context else None
+        context_str = ' Context: ' + ' ;; '.join(self.context) if self.context else ''
+        super().__init__(f'Could not verify module: {self.error}{context_str}')
+
+
+@final
+@dataclass
+class DuplicateModuleNameError(KoreClientError):
```

Comment:
Just for succinctness (and symmetry with the other module errors) I'd call this:

```suggestion
class DuplicateModuleError(KoreClientError):
```